thetis/solver2d.py

```diff
@@ -136,6 +136,8 @@ def __init__(self, mesh2d, bathymetry_2d, options=None):
 
         self.bnd_functions = {'shallow_water': {}, 'tracer': {}, 'sediment': {}}
 
+        if 'tracer_2d' in field_metadata:
+            field_metadata.pop('tracer_2d')
```

Comment:
why is this needed?

Reply:
It is needed in the tests which do convergence studies. They create a sequence of solver objects and solve on each one. If this isn't included then the 2nd solver object tries to add a tracer field that already exists when it is instantiated.

Reply:
Ok, so this could plausibly be removed when we deprecate `options.solve_tracer`. We still might need a workaround in the test but it's better to have it there rather than the solver object.